src/omv/ports/stm32/sensor.c

```diff
@@ -1105,27 +1134,34 @@ static void sensor_check_buffsize()
     MAIN_FB()->y += (window_h - MAIN_FB()->v) / 2;
     if (MAIN_FB()->x % 2) MAIN_FB()->x -= 1;
     if (MAIN_FB()->y % 2) MAIN_FB()->y -= 1;
+
+    // Pickout a good buffer count for the user.
+    framebuffer_auto_adjust_vbuffers();
 }
 
 // Stop allowing new data in on the end of the frame and let snapshot know that the frame has been
 // received. Note that DCMI_DMAConvCpltUser() is called before DCMI_IT_FRAME is enabled by
 // DCMI_DMAXferCplt() so this means that the last line of data is *always* transferred before
-// waiting_for_data is set to false.
+// calling framebuffer_writer_set_next_vbuffer().
 void HAL_DCMI_FrameEventCallback(DCMI_HandleTypeDef *hdcmi)
 {
-    waiting_for_data = false;
+    framebuffer_writer_set_next_vbuffer();
 }
 
 // This function is called back after each line transfer is complete,
 // with a pointer to the line buffer that was used. At this point the
 // DMA transfers the next line to the other half of the line buffer.
 void DCMI_DMAConvCpltUser(uint32_t addr)
 {
+    vbuffer_state_t *state = framebuffer_writer_get_next_vbuffer();
```

Comment:
```suggestion
    vbuffer_t *vbuffer = framebuffer_get_next_buffer();
```
Replace state/vbuffer in sensor.c

Reply:
Done